src/App.jsx

```diff
@@ -13,58 +15,57 @@ export const goods = [
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
-
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
+  return (
+    <main className="section container">
+      {selectedGood === null ? (
+        <h1 className="title is-flex is-align-items-center">
+          No goods selected
+        </h1>
+      ) : (
+        <h1 className="title is-flex is-align-items-center">
+          {selectedGood} is selected
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={() => setSelectedGood(null)}
+          />
+        </h1>
+      )}
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+            <tr
+              key={good}
+              data-cy="Good"
+              className={cn('', {
+                'has-background-success-light': selectedGood === good,
+              })}
             >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+              <td>
+                <button
+                  data-cy={selectedGood === good ? 'RemoveButton' : 'AddButton'}
+                  type="button"
+                  onClick={() =>
+                    setSelectedGood(selectedGood === good ? null : good)
+                  }
+                  className={cn('button', {
+                    'is-info': selectedGood === good,
+                  })}
+                >
+                  {selectedGood === good ? '-' : '+'}
+                </button>
+              </td>
+              <td data-cy="GoodTitle" className="is-vcentered">
+                {good}
+              </td>
+            </tr>
+          ))}
+        </tbody>
+      </table>
+    </main>
+  );
+};
```

Comment:
Although the code is working fine, it's a good practice to separate the logic from the rendering part. I suggest you to create a separate function for handling the onClick event. This will improve the code readability and maintainability. Here is an example of how you can do it: 

jsx
const handleOnClick = (good) => {
  setSelectedGood(selectedGood === good ? null : good);
};

<button
  data-cy={selectedGood === good ? 'RemoveButton' : 'AddButton'}
  type="button"
  onClick={() => handleOnClick(good)}
  className={cn('button', {
    'is-info': selectedGood === good,
  })}
>
  {selectedGood === good ? '-' : '+'}
</button>
```
This way, the logic is abstracted away from the rendering part, making the code cleaner and easier to read.